File: HRMS/HRMS.Web/Modules/Operations/LeaveType/LeaveTypeColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace HRMS.Operations.Columns;

[ColumnsScript("Operations.LeaveType")]
[BasedOnRow(typeof(LeaveTypeRow), CheckNames = true)]
public class LeaveTypeColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int LeaveTypeId { get; set; }
    [EditLink]
    public string LeaveTypeName { get; set; }
    public string LeaveCode { get; set; }
    public LeaveCategory LeaveCategory { get; set; }
    public decimal AnnualAllocation { get; set; }
    public bool MonthlyAccrual { get; set; }
    public bool CarryForwardAllowed { get; set; }
    public bool SandwichRuleApplicable { get; set; }
    public bool HalfDayAllowed { get; set; }
    public RecordStatus Status { get; set; }
}

File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/LeaveTypeColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { LeaveCategory } from "./LeaveCategory";
import { LeaveTypeRow } from "./LeaveTypeRow";

export interface LeaveTypeColumns {
    LeaveTypeId: Column<LeaveTypeRow>;
    LeaveTypeName: Column<LeaveTypeRow>;
    LeaveCode: Column<LeaveTypeRow>;
    LeaveCategory: Column<LeaveTypeRow>;
    AnnualAllocation: Column<LeaveTypeRow>;
    MonthlyAccrual: Column<LeaveTypeRow>;
    CarryForwardAllowed: Column<LeaveTypeRow>;
    SandwichRuleApplicable: Column<LeaveTypeRow>;
    HalfDayAllowed: Column<LeaveTypeRow>;
    Status: Column<LeaveTypeRow>;
}

export class LeaveTypeColumns extends ColumnsBase<LeaveTypeRow> {
    static readonly columnsKey = 'Operations.LeaveType';
    static readonly Fields = fieldsProxy<LeaveTypeColumns>();
}

[LeaveCategory]; // referenced types